From: John Jolly <john.jolly@gmail.com>
Date: Wed Sep 12 11:43:35 2018 -0600
Subject: Provisioning/httpd: Make plugin directory configurable at build time (#138)
Patch-mainline: Upstream
Git-repo: https://github.com/warewulf/warewulf3
Git-commit: 345526c51062a20b38daf70afd6a10105adef966
References: 

Apache2 plugins may live in different places for different distribution
vendors. Make their path configurable thru an option.

Signed-off-by: Egbert Eich <eich@suse.com>
Signed-off-by: John Jolly <jjolly@suse.com>---
 provision/configure.ac               | 11 +++++++++++
 provision/etc/warewulf-httpd.conf.in |  4 ++--
 2 files changed, 13 insertions(+), 2 deletions(-)
diff --git a/provision/configure.ac b/provision/configure.ac
index cd5e985..3060257 100644
--- a/provision/configure.ac
+++ b/provision/configure.ac
@@ -64,6 +64,17 @@ PERL_VENDORLIB="$force_prefix$PERL_VENDORLIB"
 AC_MSG_RESULT($PERL_VENDORLIB)
 AC_SUBST(PERL_VENDORLIB)
 
+apache_moddir="modules"
+AC_MSG_CHECKING([for Apache2 module path])
+AC_ARG_WITH(apache2moddir, [  --with-apache2moddir=path  Manually specify where the Apache2 modules live], [
+if test -n "$withval" -a "x$withval" != "xno" ; then
+   apache_moddir="$withval"
+fi
+])
+AC_MSG_RESULT($apache_moddir)
+AC_SUBST(apache_moddir)
+
+
 #AC_CHECK_LIB(fuse, fuse_main, , [
 #AC_MSG_ERROR([Fatal:  Fuse libraries not found.])
 #])
diff --git a/provision/etc/warewulf-httpd.conf.in b/provision/etc/warewulf-httpd.conf.in
index c0d3521..b6644c1 100644
--- a/provision/etc/warewulf-httpd.conf.in
+++ b/provision/etc/warewulf-httpd.conf.in
@@ -1,8 +1,8 @@
 <IfModule !perl_module>
-    LoadModule perl_module modules/mod_perl.so
+    LoadModule perl_module @apache_moddir@/mod_perl.so
 </IfModule>
 <IfModule !version_module>
-    LoadModule version_module modules/mod_version.so
+    LoadModule version_module @apache_moddir@/mod_version.so
 </IfModule>
 PerlSwitches -w
 PerlSwitches -T
